samtranslator/validator/sam_schema/definitions/api.json

```diff
@@ -0,0 +1,789 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "additionalProperties": false,
+  "type": "object",
+  "definitions": {
+    "AWS::Serverless::Api": {
+      "if": {
+        "properties": {
+          "Type": {
+            "const": "AWS::Serverless::Api"
+          }
+        },
+        "required": [
+          "Type"
+        ]
+      },
+      "then": {
+        "additionalProperties": false,
+        "properties": {
+          "Condition": {
+            "$ref": "file:common.json#definitions/Condition"
+          },
+          "DependsOn": {
+            "$ref": "file:common.json#definitions/DependsOn"
+          },
+          "Metadata": {
+            "type": "object"
+          },
+          "Properties": {
+            "additionalProperties": false,
+            "properties": {
+              "AccessLogSetting": {
+                "$ref": "file:cfn.json#definitions/AWS::ApiGateway::Stage.AccessLogSetting"
```

Comment:
Do we want to handle CFN objects?
Are we able to use [CFN schemas](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/resource-type-schemas.html) and link those fields to here with pointers?

Reply:
I'm using these files as a base for building the schema:
- https://raw.githubusercontent.com/awslabs/goformation/master/schema/sam.schema.json
- https://raw.githubusercontent.com/awslabs/goformation/master/schema/cloudformation.schema.json

That's where the names come from.
Maybe we could drop the `cfn.json` file, rename these common properties and put them in `common.json`.
`AWS::ApiGateway::Stage.AccessLogSetting` would become `AccessLogSetting` for example.

Reply:
I think I was initially thinking of putting in this file the definitions for which we directly link to the Cloud Formation documentation in our SAM spec doc.

Reply:
@jfuss Your opinion on this?
As we link to the CFN doc in our doc, the naming would be more consistent with the doc:
- https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-api.html

Links to
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-stage.html#cfn-apigateway-stage-accesslogsetting
- https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-stage-accesslogsetting.html